packages/cli/src/commands/split/index.ts

```diff
@@ -237,49 +242,62 @@ function gatherComponentsFiles(
   componentsFiles[componentType][componentName] = { inherits, filename };
 }
 
-function iteratePaths(
-  openapi: Oas3Definition,
-  pathsDir: string,
-  openapiDir: string
+function iteratePathItem(
+  openapi: any,
+  openapiDir: string,
+  componentsFiles: object,
+  rootDocObj: string,
+  customDir?: string,
 ) {
-  const { paths } = openapi;
-  if (paths) {
-    for (const oasPath of Object.keys(paths)) {
-      const pathFile = path.join(pathsDir, pathToFilename(oasPath)) + '.yaml';
-      const pathData: Oas3PathItem = paths[oasPath] as Oas3PathItem;
+  const docObj = rootDocObj === PATHS ? openapi.paths : (openapi[WEBHOOKS] ? openapi[WEBHOOKS] : openapi[xWEBHOOKS]);
+
+  if (docObj) {
+    const checkDir = customDir || path.join(openapiDir, rootDocObj);
+    fs.mkdirSync(checkDir, { recursive: true });
+
+    for (const oasPath of Object.keys(docObj)) {
```

Comment:
I know it's out of this issue, but I would like to refactor this function a bit, it includes 3 nested `for` (